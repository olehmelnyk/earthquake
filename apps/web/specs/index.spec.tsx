import React from 'react';
import { render, screen } from '@testing-library/react';
import { describe, it, expect, vi } from 'vitest';
import Page from '../src/app/page';

// Mock the Apollo provider
vi.mock('../src/app/apollo-provider', () => ({
  ApolloWrapper: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="apollo-wrapper">{children}</div>
  )
}));

// Mock the UI components
vi.mock('@earthquake/ui', () => ({
  ThemeProvider: ({ children }: { children: React.ReactNode }) => (
    <div data-testid="theme-provider">{children}</div>
  ),
  Toaster: () => <div data-testid="toaster">Toaster</div>
}));

// Mock the Dashboard component
vi.mock('../src/app/components/Dashboard', () => ({
  Dashboard: () => <div data-testid="dashboard">Dashboard Component</div>
}));

describe('HomePage', () => {
  it('should render successfully', () => {
    render(<Page />);

    // Check if Dashboard component is rendered
    const dashboard = screen.getByTestId('dashboard');
    expect(dashboard).toBeInTheDocument();
    expect(dashboard).toHaveTextContent('Dashboard Component');
  });

  it('should match snapshot', () => {
    const { container } = render(<Page />);
    expect(container).toMatchSnapshot();
  });
});
