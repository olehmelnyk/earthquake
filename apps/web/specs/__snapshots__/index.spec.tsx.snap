// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomePage > should match snapshot 1`] = `
<div>
  <div
    data-testid="dashboard"
  >
    Dashboard Component
  </div>
</div>
`;
